var/spack/repos/builtin/packages/faiss/package.py

```diff
@@ -0,0 +1,131 @@
+
+    # patch for v1.6.3
+    # for v1.6.3, GPU build has a bug (two files need to be deleted)
+    # https://github.com/facebookresearch/faiss/issues/1159
+    # also, some include paths in gpu/tests/Makefile are missing
+    patch('fixes-in-v1.6.3.patch', when='@1.6.3')
+
+    def configure_args(self):
+        args = []
+        args.extend(self.with_or_without('cuda', activation_value='prefix'))
+        return args
+
+    def build(self, spec, prefix):
+
+        make()
+
+        if '+python' in self.spec:
+            make('-C', 'python')
+
+        # CPU tests
+        if '+tests' in self.spec:
+            with working_dir('tests'):
+                make('gtest')
+                make('tests')
+
+        # GPU tests
+        if '+tests' in self.spec and '+cuda' in self.spec:
+            with working_dir(os.path.join('gpu', 'test')):
+                make('gtest')
+                make('build')                       # target added by the patch
+                make('demo_ivfpq_indexing_gpu')
+
+    def install(self, spec, prefix):
+
+        make('install')
+
+        if '+python' in self.spec:
+            with working_dir('python'):
+                setup_py('install', '--prefix=' + prefix,
+                         '--single-version-externally-managed', '--root=/')
+
+        if not os.path.isdir(self.prefix.bin):
+            os.makedirs(self.prefix.bin)
+
+        def _prefix_and_install(file):
+            os.system('mv {} faiss_{}'.format(file, file))
+            install('faiss_{}'.format(file), self.prefix.bin)
+
+        # CPU tests
+        if '+tests' in self.spec:
+            with working_dir('tests'):
+                # rename the exec to keep consistent with gpu tests
+                os.system('mv tests TestCpu')
+                _prefix_and_install('TestCpu')
+
+        # GPU tests
+        if '+tests' in self.spec and '+cuda' in self.spec:
+            with working_dir(os.path.join('gpu', 'test')):
+                _prefix_and_install('TestGpuIndexFlat')
+                _prefix_and_install('TestGpuIndexBinaryFlat')
+                _prefix_and_install('TestGpuIndexIVFFlat')
+                _prefix_and_install('TestGpuIndexIVFPQ')
+                _prefix_and_install('TestGpuMemoryException')
+                _prefix_and_install('TestGpuSelect')
+                _prefix_and_install('demo_ivfpq_indexing_gpu')
+
+    @run_after('configure')
+    def _fix_makefile(self):
+
+        # spack injects its own optimization flags
+        makefile = FileFilter('makefile.inc')
+        makefile.filter('CPUFLAGS     = -mavx2 -mf16c',
+                        '#CPUFLAGS     = -mavx2 -mf16c')
+
+    def setup_run_environment(self, env):
+        pv = sys.version_info
+        env.prepend_path('PYTHONPATH',
+                         join_path(self.prefix.lib,
+                                   'python{}.{}'.format(pv[0], pv[1]),
+                                   'site-packages'))
```

Comment:
You should be able to use `site_packages_dir` here, it's set by the `python` package.